kusto/kql/query_parameters_test.go

```diff
@@ -0,0 +1,33 @@
+package kql
+
+import (
+	"fmt"
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func TestQueryParameters(t *testing.T) {
+	tests := []struct {
+		name     string
+		b        Builder
+		qp       *StatementQueryParameters
+		expected string
+	}{
+		{"Test empty", NewStatementBuilder(""), NewStatementQueryParameters(), "\n"},
+		{"Test add string", NewStatementBuilder(""), NewStatementQueryParameters().AddString("foo", "string", "bar"), "declare query_parameters(foo:string);\n"},
+		{"Test add identifiers", // test might fail at times due to the pseudo-random nature of map that will sometimes change the order of the declaration string.
```

Comment:
No, don't make tests that might fail. Have the check be more clever.

Reply:
@yogilad you said it's not worth the time and just leave a note. what do we think we should do?

Reply:
Spoke to Yochai - he said that the test shouldn't be flakey. Just check the substring.